### ISSUE TITLE
Document that Manifold doesn't actually use its 'dim' template argument

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#2256]
@luca-heltai -- is it correct that neither `Manifold` not `FlatManifold` actually use their `dim` template argument? My interpretation is that they both just describe general manifolds for which it is not important what their dimensionality actually is. (This does not imply that this is not an issue for derived classes.) In other words, these two classes only have these template arguments because that's what we've been doing with all other classes.

If you agree with this interpretation, I'll add a blurb to this effect into the documentation.

**Comment by luca-heltai** [comment]
Yes, that is correct. I think this is something we should address when we remove backward compatibility from Manifold. In my opinion, we should remove the additional template parameter from this class. When I implemented this the first time, I did not manage to make it work without the `dim` template. I don't remember any more what went wrong. We can look into this next week...

**Comment by bangerth** [comment]
We can think about it. I don't mind leaving the argument. It's not that leaving it in makes our lives actively more complicated.

**Comment by bangerth** [comment]
See also #2317.

**Comment by drwells** [comment]
I remembered this issue and, after looking at Manifold much more closely this week, it looks like we do use `dim` to define `FaceVertexNormals`. @luca-heltai This looks like the only way to get normal vectors out of manifolds: should we keep the related functions and close this issue?
